src/Uno.UI/UI/Xaml/ElementStub.cs

```diff
@@ -194,6 +194,18 @@ private void MaterializeInner()
 				{
 					_isMaterializing = true;
 
+					// Before swapping out the ElementStub, we set the inherited DC locally so that when removed from the
+					// visual tree, the inherited DC is kept intact. This is important in case of setting x:Load with an
+					// x:Bind. The binding we generate from the x:Bind will only listen to property changes while the DC
+					// is present. This is definitely not what happens on WinUI, but considering our implementation of
+					// x:Bind and ElementStub differ completely from WinUI's, this is acceptable for now.
+					// https://github.com/unoplatform/uno/issues/18509
+					if ((this as IDependencyObjectStoreProvider).Store.GetPropertyDetails(DataContextProperty)
+						.CurrentHighestValuePrecedence > DependencyPropertyValuePrecedences.Local)
+					{
+						this.SetValue(DataContextProperty, DataContext, DependencyPropertyValuePrecedences.Local);
+					}
+
```

Comment:
For the record, I _really_ don't like this, but I don't see a better way of fixing this with the way x:Bind is currently implemented.